pkg/cliplugins/workspace/plugin/kubeconfig.go

```diff
@@ -805,10 +805,7 @@ func (o *TreeOptions) Run(ctx context.Context) error {
 
 func (o *TreeOptions) populateBranch(ctx context.Context, tree treeprint.Tree, name logicalcluster.Name) error {
 	results, err := o.kcpClusterClient.Cluster(name).TenancyV1beta1().Workspaces().List(ctx, metav1.ListOptions{})
-	if err != nil {
```

Comment:
It's not valid to access `results` when `err != nil`. This patch makes that happen when `apierrors.IsNotFound(err) == true`. Can you move the `tree.AddBranch()` bits to above this fetch instead?

Reply:
yes, you right. Rushed solution.